drivers/spi/spi_ll_stm32.c

```diff
@@ -803,7 +809,8 @@ static int transceive_dma(const struct device *dev,
 	}
 
 #ifdef CONFIG_SOC_SERIES_STM32H7X
-	if ((tx_bufs != NULL && !spi_buf_set_in_nocache(tx_bufs)) ||
+	if ((tx_bufs != NULL && !is_null_buffer(tx_bufs) &&
+		!spi_buf_set_in_nocache(tx_bufs)) ||
```

Comment:
I have re-read the SPI header code docs. This is what it exactly says about null buffers:

```
/**
 * @brief SPI buffer structure
 *
 * @param buf is a valid pointer on a data buffer, or NULL otherwise.
 * @param len is the length of the buffer or, if buf is NULL, will be the
 *    length which as to be sent as dummy bytes (as TX buffer) or
 *    the length of bytes that should be skipped (as RX buffer).
 */
struct spi_buf {
	void *buf;
	size_t len;
};
```
Any buffer in the buf_set could be null and then, dummy bytes should be sent or ignored. Therefore, I would change this function by:

```
static bool is_dummy_buffer(const struct spi_buf* buf)
{
	return buf->buf == NULL;
}
...
static bool spi_buf_set_in_nocache(const struct spi_buf_set *bufs)
{
	for (size_t i = 0; i < bufs->count; i++) {
		const struct spi_buf *buf = &bufs->buffers[i];

		if (!is_dummy_buffer(buf) &&
			!buf_in_nocache((uintptr_t)buf->buf, buf->len)) {
			return false;
		}
	}
	return true;
}
``` 
and leave this as it was:
```
#ifdef CONFIG_SOC_SERIES_STM32H7X
	if ((tx_bufs != NULL && !spi_buf_set_in_nocache(tx_bufs)) ||
		(rx_bufs != NULL && !spi_buf_set_in_nocache(rx_bufs))) {
		return -EFAULT;
	}
#endif /* CONFIG_SOC_SERIES_STM32H7X */
```

Reply:
Hi again, don't worry about this because I have included this change in #61265